hollow/src/main/java/com/netflix/hollow/core/read/engine/object/HollowObjectTypeReadState.java

```diff
@@ -265,39 +696,43 @@ public void setFieldSpecificSamplingDirector(HollowFilterConfig fieldSpec, Hollo
     public void ignoreUpdateThreadForSampling(Thread t) {
         sampler.setUpdateThread(t);
     }
-    
+
     HollowObjectTypeDataElements[] currentDataElements() {
-        HollowObjectTypeDataElements currentDataElements[] = new HollowObjectTypeDataElements[shards.length];
-        
-        for(int i=0;i<shards.length;i++)
-            currentDataElements[i] = shards[i].currentDataElements();
-        
-        return currentDataElements;
+        final HollowObjectTypeReadStateShard[] shards = this.shardsVolatile.shards;
+        return Arrays.stream(shards)
+                .map(shard -> shard.dataElements)
+                .toArray(HollowObjectTypeDataElements[]::new);
```

Comment:
The stream approach feels slightly harder to read, that might be my own bias.

Reply:
Sure, reworked it